src/osemosys2iamc/resultify.py

```diff
@@ -29,90 +31,96 @@ def read_file(filename) -> pd.DataFrame:
 
     return df
 
-def filter_var_cost(df: pd.DataFrame, technologies: List) -> pd.DataFrame:
-    """Return rows that match the ``technologies``
-    """
-    mask = df['TECHNOLOGY'].str.contains(technologies[0])
+def filter_regex(df: pd.DataFrame, patterns: List[str], column: str) -> pd.DataFrame:
+    """Generic filtering of rows based on columns that match a list of patterns
 
-    df['REGION'] = df['TECHNOLOGY'].str[0:2]
-    df = df.drop(columns=['TECHNOLOGY', 'MODE_OF_OPERATION'])
+    This function returns the rows where the values in a ``column`` match the
+    list of regular expression ``patterns``
+    """
+    masks = [df[column].str.match(p) for p in patterns]
+    return pd.concat([df[mask] for mask in masks])
+
+def filter_fuels(df: pd.DataFrame, fuels: List[str]) -> pd.DataFrame:
+    """Returns rows which match list of regex patterns in ``technologies``
+
+    Parameters
+    ----------
+    df: pd.DataFrame
+        The input data
+    fuels: List[str]
+        List of regex patterns
+    """
+    return filter_regex(df, fuels, 'FUEL')
 
-    df = df[mask]
+def filter_technologies(df: pd.DataFrame, technologies: List[str]) -> pd.DataFrame:
+    """Returns rows which match list of regex patterns in ``technologies``
 
-    return df[df.VALUE != 0]
+    Parameters
+    ----------
+    df: pd.DataFrame
+        The input data
+    technologies: List[str]
+        List of regex patterns
+    """
+    return filter_regex(df, technologies, 'TECHNOLOGY')
 
 def filter_fuel(df: pd.DataFrame, technologies: List, fuels: List) -> pd.DataFrame:
```

Comment:
Should we rename this function? Might get easily confused with the above `filter_fuels`.

Reply:
Good idea.